packages/driver/src/cypress/error_messages.coffee

```diff
@@ -1074,20 +1071,50 @@ module.exports = {
       URL from the `options` object: {{optionsUrl}}
       URL from the `url` parameter: {{url}}
     """
-    cannot_visit_2nd_domain: """
-      #{cmd('visit')} failed because you are attempting to visit a second unique domain.
+    cannot_visit_different_protocol: """
+      #{cmd('visit')} failed because you are attempting to visit a URL that is of a different origin.
+
+      The new URL is considered different origin because the 'protocol' is different.
+
+      You may only 'cy.visit()' same origin URLs within a single test.
+
+      The previous domain you visited was: '{{previousDomain}}'
+
+      You're attempting to visit this URL: '{{attemptedDomain}}'
+
+      You may need to restructure some of your test code to avoid this problem.
+
+      https://on.cypress.io/cannot-visit-different-origin-domain
+    """
+    cannot_visit_different_port: """
+      #{cmd('visit')} failed because you are attempting to visit a URL that is of a different origin.
 
-      You may only visit a single unique domain per test.
+      The new URL is considered different origin because the 'port' is different.
```

Comment:
```suggestion
      The new URL is considered a different origin because the port is different.
```

Reply:
I know that "different origin" is supposed to be read with the same inflection as "same-origin", but it doesn't really read like that to me, what do you think about "a different origin" instead?